🌐 [i18n-KO] Translated `guides/community.md` to Korean (#2126)
huggingface/huggingface_hub: docs/source/ko/guides/community.md

<!--⚠️ Note that this file is in Markdown but contain specific syntax for our doc-builder (similar to MDX) that may not be
rendered properly in your Markdown viewer.
-->

> # 토론 및 풀 리퀘스트와 상호작용하기[[interact-with-discussions-and-pull-requests]]

Reviewer: ```suggestion
# Discussions 및 Pull Requests와 상호작용하기[[interact-with-discussions-and-pull-requests]]
```
토론 -> Discussions, 풀 리퀘스트 -> Pull Requests로 원문 그대로 두었습니다

Reviewer 2: Discussions 앞에 [?] 같은 깨진 글자가 보입니다.

Author: ```suggestion
# Discussions 및 Pull Requests를 이용하여 상호작용하기[[interact-with-discussions-and-pull-requests]]
```

> `huggingface_hub` 라이브러리는 허브의 풀 리퀘스트 및 토론과 상호작용할 수 있는 Python 인터페이스를 제공합니다.

Reviewer: ```suggestion
`huggingface_hub` 라이브러리는 Hub의 Pull Requests 및 Discussions와 상호작용할 수 있는 Python 인터페이스를 제공합니다.
```
허브 -> Hub, 풀 리퀘스트 -> Pull Requests, 토론 -> Discussions로 변경했습니다

> [전용 문서 페이지](https://huggingface.co/docs/hub/repositories-pull-requests-discussions)를 방문하여 Hub의 토론과 풀 리퀘스트가 무엇이고 어떻게 작동하는지 자세히 살펴보세요.

Reviewer: ```suggestion
[전용 문서 페이지](https://huggingface.co/docs/hub/repositories-pull-requests-discussions)를 방문하여 Hub의 Discussions와 Pull Requests가 무엇이고 어떻게 작동하는지 자세히 살펴보세요.
```
풀 리퀘스트 -> Pull Requests, 토론 -> Discussions로 변경했습니다

> ## 허브에서 토론 및 풀 리퀘스트 가져오기[[retrieve-discussions-and-pull-requests-from-the-hub]]

Reviewer: ```suggestion
## Hub에서 Discussions 및 Pull Requests 가져오기[[retrieve-discussions-and-pull-requests-from-the-hub]]
```
허브 -> Hub, 풀 리퀘스트 -> Pull Requests, 토론 -> Discussions로 변경했습니다

> `HfApi` 클래스를 사용하면 지정된 리포지토리에 대한 토론 및 풀 리퀘스트를 검색할 수 있습니다:

Reviewer: ```suggestion
`HfApi` 클래스를 사용하면 지정된 리포지토리에 대한 Discussions 및 Pull Requests를 검색할 수 있습니다:
```
풀 리퀘스트 -> Pull Requests, 토론 -> Discussions로 변경했습니다

```python
>>> from huggingface_hub import get_repo_discussions
>>> for discussion in get_repo_discussions(repo_id="bigscience/bloom"):
...     print(f"{discussion.num} - {discussion.title}, pr: {discussion.is_pull_request}")

# 11 - Add Flax weights, pr: True
# 10 - Update README.md, pr: True
# 9 - Training languages in the model card, pr: True
# 8 - Update tokenizer_config.json, pr: True
# 7 - Slurm training script, pr: False
[...]
```

> `HfApi.get_repo_discussion`은 작성자, 유형(풀 리퀘스트 또는 토론) 및 상태(`열림` 또는 `닫힘`)별로 필터링을 지원합니다:

Reviewer: ```suggestion
`HfApi.get_repo_discussion`은 작성자, 유형(Pull Requests 또는 Discussion) 및 상태(`open` 또는 `closed`)별로 필터링을 지원합니다:
```
풀 리퀘스트 -> Pull Requests, 토론 -> Discussions로 변경했습니다
열림 -> open, 닫힘 -> closed로 원문으로 남겼습니다

```python
>>> from huggingface_hub import get_repo_discussions
>>> for discussion in get_repo_discussions(
...    repo_id="bigscience/bloom",
...    author="ArthurZ",
...    discussion_type="pull_request",
...    discussion_status="open",
... ):
...     print(f"{discussion.num} - {discussion.title} by {discussion.author}, pr: {discussion.is_pull_request}")

# 19 - Add Flax weights by ArthurZ, pr: True
```

> `HfApi.get_repo_discussions`는 [`Discussion`] 객체를 생성하는 [제너레이터](https://docs.python.org/3.7/howto/functional.html#generators)를 반환합니다. 모든 토론들을 하나의 목록으로 가져오려면 다음을 실행합니다:

Reviewer 3: ```suggestion
`HfApi.get_repo_discussions`는 [`Discussion`] 객체를 생성하는 [제너레이터](https://docs.python.org/3.7/howto/functional.html#generators)를 반환합니다. 모든 토론을 하나의 목록으로 가져오려면 다음을 실행합니다:
```
'모든' 자체가 복수이기에, 접미사 '-들'을 지웠습니다!

Author: 리뷰 감사합니다! 관련 사항이 궁금하여 찾아보니, [국립국어원에서 답변한 내용](https://www.korean.go.kr/front/onlineQna/onlineQnaView.do?mn_id=&qna_seq=253197&pageIndex=1)을 참고하였을 때, 모든 뒤에 복수 명사가 나와도 괜찮은 것 같습니다..!

Reviewer 3: 헉 몰랐네요.. 그러면 이외에는 특별히 없는거 같습니다!!

> `HfApi.get_repo_discussions`는 [`Discussion`] 객체를 생성하는 [제너레이터](https://docs.python.org/3.7/howto/functional.html#generators)를 반환합니다. 모든 토론들을 하나의 목록으로 가져오려면 다음을 실행합니다:

Reviewer: ```suggestion
`HfApi.get_repo_discussions`는 [`Discussion`] 객체를 생성하는 [생성자](https://docs.python.org/3.7/howto/functional.html#generators)를 반환합니다. 모든 Discussions를 하나의 리스트로 가져오려면 다음을 실행합니다:
```
딥러닝 용어사전에 따르면 generator가 생성자이기는 한데, 또 제너레이터로 번역하는 게 맞을 것 같기도 하네요...? 이 부분은 확인이 필요할 것 같아요...!
토론 -> Discussions으로 변경했습니다
목록 -> 리스트로 변경했습니다

```python
>>> from huggingface_hub import get_repo_discussions
>>> discussions_list = list(get_repo_discussions(repo_id="bert-base-uncased"))
```

> [`HfApi.get_repo_discussions`]가 반환하는 [`Discussion`] 객체에는 토론 또는 풀 리퀘스트에 대한 개략적인 개요가 포함되어 있습니다. [`HfApi.get_discussion_details`]를 사용하여 더 자세한 정보를 얻을 수도 있습니다:

Reviewer: ```suggestion
[`HfApi.get_repo_discussions`]가 반환하는 [`Discussion`] 객체에는 Discussions 또는 Pull Request에 대한 개략적인 개요가 포함되어 있습니다. [`HfApi.get_discussion_details`]를 사용하여 더 자세한 정보를 얻을 수도 있습니다:
```
풀 리퀘스트 -> Pull Requests, 토론 -> Discussions로 변경했습니다

```python
>>> from huggingface_hub import get_discussion_details

>>> get_discussion_details(
...     repo_id="bigscience/bloom-1b3",
...     discussion_num=2
... )
DiscussionWithDetails(
    num=2,
    author='cakiki',
    title='Update VRAM memory for the V100s',
    status='open',
    is_pull_request=True,
    events=[
        DiscussionComment(type='comment', author='cakiki', ...),
        DiscussionCommit(type='commit', author='cakiki', summary='Update VRAM memory for the V100s', oid='1256f9d9a33fa8887e1c1bf0e09b4713da96773a', ...),
    ],
    conflicting_files=[],
    target_branch='refs/heads/main',
    merge_commit_oid=None,
    diff='diff --git a/README.md b/README.md\nindex a6ae3b9294edf8d0eda0d67c7780a10241242a7e..3a1814f212bc3f0d3cc8f74bdbd316de4ae7b9e3 100644\n--- a/README.md\n+++ b/README.md\n@@ -132,7 +132,7 [...]',
)
```

> [`HfApi.get_discussion_details`]는 토론 또는 풀 리퀘스트에 대한 자세한 정보가 포함된 [`Discussion`]의 하위 클래스인 [`DiscussionWithDetails`] 객체를 반환합니다. 해당 정보는 [`DiscussionWithDetails.events`]를 통해 토론의 모든 댓글, 상태 변경 및 이름 변경을 포함하고 있습니다.

Reviewer: ```suggestion
[`HfApi.get_discussion_details`]는 Discussion 또는 Pull Request에 대한 자세한 정보가 포함된 [`Discussion`]의 하위 클래스인 [`DiscussionWithDetails`] 객체를 반환합니다. 해당 정보는 [`DiscussionWithDetails.events`]를 통해 Discussion의 모든 댓글, 상태 변경 및 이름 변경을 포함하고 있습니다.
```
풀 리퀘스트 -> Pull Requests, 토론 -> Discussion로 변경했습니다

> 풀 리퀘스트의 경우, [`DiscussionWithDetails.diff`]를 통해 원시 git diff를 검색할 수 있습니다. 풀 리퀘스트의 모든 커밋은 [`DiscussionWithDetails.events`]에 나열됩니다.

Reviewer: ```suggestion
Pull Request의 경우, [`DiscussionWithDetails.diff`]를 통해 원시 git diff를 검색할 수 있습니다. Pull Request의 모든 커밋은 [`DiscussionWithDetails.events`]에 나열됩니다.
```
풀 리퀘스트 -> Pull Request로 수정했습니다.

> ## 프로그래밍 방식으로 토론 또는 풀 리퀘스트를 생성하고 수정하기[[create-and-edit-a-discussion-or-pull-request-programmatically]]

Reviewer: ```suggestion
## 프로그래밍 방식으로 Discussion 또는 Pull Request를 생성하고 수정하기[[create-and-edit-a-discussion-or-pull-request-programmatically]]
```
풀 리퀘스트 -> Pull Request, 토론 -> Discussion로 변경했습니다

> [`HfApi`] 클래스는 토론 및 풀 리퀘스트를 생성하고 수정하는 방법도 제공합니다.

Reviewer: ```suggestion
[`HfApi`] 클래스는 Discussions 및 Pull Requests를 생성하고 수정하는 방법도 제공합니다.
```
풀 리퀘스트 -> Pull Request, 토론 -> Discussion로 변경했습니다

> 토론이나 풀 리퀘스트를 만들고 편집하려면 [접근 토큰](https://huggingface.co/docs/hub/security-tokens)이 필요합니다.

Reviewer: ```suggestion
Discussions와 Pull Requests를 만들고 편집하려면 [접근 토큰](https://huggingface.co/docs/hub/security-tokens)이 필요합니다.
```
풀 리퀘스트 -> Pull Request, 토론 -> Discussion로 변경했습니다

Hub의 리포지토리에 변경 사항을 제안하는 가장 간단한 방법은 [`create_commit`] API를 사용하는 것입니다. `create_pr` 매개변수를 `True`로 설정하기만 하면 됩니다. 이 매개변수는 [`create_commit`]을 래핑하는 다른 함수에서도 사용할 수 있습니다:

    * [`upload_file`]
    * [`upload_folder`]
    * [`delete_file`]
    * [`delete_folder`]
    * [`metadata_update`]

```python
>>> from huggingface_hub import metadata_update

>>> metadata_update(
...     repo_id="username/repo_name",
...     metadata={"tags": ["computer-vision", "awesome-model"]},
...     create_pr=True,
... )
```

> 리포지토리에 대한 토론(또는 풀 리퀘스트)을 만들려면 [`HfApi.create_discussion`](또는 [`HfApi.create_pull_request`])을 사용할 수도 있습니다.

Reviewer: ```suggestion
리포지토리에 대한 Discussion(또는 Pull Request)을 만들려면 [`HfApi.create_discussion`](또는 [`HfApi.create_pull_request`])을 사용할 수도 있습니다.
```
풀 리퀘스트 -> Pull Request, 토론 -> Discussion로 변경했습니다

> 이 방법으로 풀 리퀘스트를 열면 로컬에서 변경 작업을 해야 하는 경우에 유용할 수 있습니다. 이 방법으로 열린 풀 리퀘스트는 `"초안"` 모드가 됩니다.

Reviewer: ```suggestion
이 방법으로 Pull Request를 열면 로컬에서 변경 작업을 해야 하는 경우에 유용할 수 있습니다. 이 방법으로 열린 Pull Request는 `"draft"` 모드가 됩니다.
```
풀 리퀘스트 -> Pull Request, 초안 -> draft로 변경했습니다.

```python
>>> from huggingface_hub import create_discussion, create_pull_request

>>> create_discussion(
...     repo_id="username/repo-name",
...     title="Hi from the huggingface_hub library!",
...     token="<insert your access token here>",
... )
DiscussionWithDetails(...)

>>> create_pull_request(
...     repo_id="username/repo-name",
...     title="Hi from the huggingface_hub library!",
...     token="<insert your access token here>",
... )
DiscussionWithDetails(..., is_pull_request=True)
```

> 풀 리퀘스트 및 토론 관리는 전적으로 [`HfApi`] 클래스로 할 수 있습니다. 예를 들어:

Reviewer: ```suggestion
Pull Requests 및 Discussions 관리는 전적으로 [`HfApi`] 클래스로 할 수 있습니다. 예를 들어:
```
풀 리퀘스트 -> Pull Request, 토론 -> Discussion로 변경했습니다

    * 댓글을 추가하려면 [`comment_discussion`]
    * 댓글을 수정하려면 [`edit_discussion_comment`]

>     * 토론 또는 풀 리퀘스트의 이름을 바꾸려면 [`rename_discussion`]

Reviewer: ```suggestion
    * Discussion 또는 Pull Request의 이름을 바꾸려면 [`rename_discussion`]
```
풀 리퀘스트 -> Pull Request, 토론 -> Discussion로 변경했습니다

>     * 토론/풀 리퀘스트를 열거나 닫으려면 [`change_discussion_status`]

Reviewer: ```suggestion
    * Discussion / Pull Request를 열거나 닫으려면 [`change_discussion_status`]
```
풀 리퀘스트 -> Pull Request, 토론 -> Discussion로 변경했습니다

>     * 풀 리퀘스트를 병합하려면 [`merge_pull_request`]를 사용합니다.

Reviewer: ```suggestion
    * Pull Request를 병합하려면 [`merge_pull_request`]를 사용합니다.
```
풀 리퀘스트 -> Pull Request로 변경했습니다

> 사용 가능한 모든 메서드에 대한 전체 참조는 [`HfApi`] 문서 페이지를 참조하세요.

Reviewer: ```suggestion
사용 가능한 모든 메소드에 대한 전체 참조는 [`HfApi`] 문서 페이지를 참조하세요.
```
glossary에 따라 메서드 -> 메소드로 변경했습니다.

> ## 풀 리퀘스트에 변경 사항 푸시[[push-changes-to-a-pull-request]]

Reviewer: ```suggestion
## Pull Request에 변경 사항 푸시[[push-changes-to-a-pull-request]]
```
풀 리퀘스트 -> Pull Request로 변경했습니다

*곧 공개됩니다!*

## 참고 항목[[see-also]]

> 더 자세한 내용은 [토론 및 풀 리퀘스트](../package_reference/community)와 [hf_api](../package_reference/hf_api) 문서 페이지를 참조하세요.

Reviewer: ```suggestion
더 자세한 내용은 [Discussions 및 Pull Requests](../package_reference/community)와 [hf_api](../package_reference/hf_api) 문서 페이지를 참조하세요.
```
풀 리퀘스트 -> Pull Request, 토론 -> Discussion로 변경했습니다
